user_guide_src/source/outgoing/response/022.php

```diff
@@ -1,4 +1,4 @@
 <?php
 
 $response->setLastModified(date('D, d M Y H:i:s'));
-$response->setLastModified(DateTime::createFromFormat('u', $time));
+$response->setLastModified(DateTime::createFromFormat('!u', $time));
```

Comment:
This change is incorrect.

```
php > var_dump(DateTime::createFromFormat('u', 654321));
php shell code:1:
class DateTime#1 (3) {
  public $date =>
  string(26) "2022-09-04 00:00:00.654321"
  public $timezone_type =>
  int(3)
  public $timezone =>
  string(3) "UTC"
}
```
```
php > var_dump(DateTime::createFromFormat('!u', 654321));
php shell code:1:
class DateTime#1 (3) {
  public $date =>
  string(26) "1970-01-01 00:00:00.654321"
  public $timezone_type =>
  int(3)
  public $timezone =>
  string(3) "UTC"
}
```

Reply:
The sample code doesn't make sense to begin with.

Reply:
Based on the docs, this is expected.

Reply:
What do you mean?

Neither of the following seems to have a use case. What do we specify by $time?
```php
$response->setLastModified(DateTime::createFromFormat('u', $time));
$response->setLastModified(DateTime::createFromFormat('!u', $time));
```

Reply:
> Resets all fields (year, month, day, hour, minute, second, fraction and timezone information) to zero-like values ( 0 for hour, minute, second and fraction, 1 for month and day, 1970 for year and UTC for timezone information)

This means that since the format lacks any of "Y, m, d, H, i, s", the ! resets them.


I think the format should have been the capital U.

Reply:
I think you're correct. `u` generates _only_ the microseconds segment, so: `2022-09-10 07:37:24.5238` yields `5238`. This was supposed to be `U` for an actual full timestamp.

Reply:
There is no microseconds in the Last-Modified header:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified

The sample code should be like the following:
```php
$response->setLastModified(DateTime::createFromFormat('U', $timestamp));
```
And it is the same as:
```php
$response->setLastModified(DateTime::createFromFormat('!U', $timestamp));
```

Reply:
I sent a PR #6526

Reply:
I think by definition `!U` is redundant.